1. Building a To-Do List Without Framework/submission/index.js

```diff
@@ -0,0 +1,108 @@
+                </div>
+                             `;
+                return elem;
+            }
+
+            function renderList() {
+                const listElement = document.getElementById("todo-list");
+
+                listElement.innerHTML = "";
+
+                for (let i = 0; i < todoList.length; i++) {
+                    const todoItem = todoElement(todoList[i], i);
+                    listElement.appendChild(todoItem);
+                }
+
+                let alert = `
+                <div class="text-center alert ${
+                    isAdded
+                        ? "alert-success"
+                        : "alert-danger"
+                } alert-dismissible fade show shadow border-0" role="alert"  data-aos="fade-down">
+                        <strong>${
+                            isAdded
+                                ? "Congraluations, Your item has been added."
+                                : "Your item has been deleted."
+                        }
+                        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+                    </div>`;
+                $("#confirmation").html(alert);
+            }
+
+            document
+                .getElementById("todo-form")
+                .addEventListener("submit", function (event) {
+                    event.preventDefault();
+                    const itemName = event.target["todo-input"].value;
+                    if (itemName) {
+                        todoList.push(generateTodoItem(itemName));
+                        isAdded = true;
+                        renderList();
+                        event.target.reset();
+                    }
+                });
+
+            function markItem(event) {
+                const itemId = event.target.parentElement.id; // getting the id of the parent element to search in the array
+                todoList.some(function (todoItem) {
+                    if (todoItem.id === itemId) {
+                        todoItem.isDone = !todoItem.isDone;
+                        renderList();
+                        return true;
+                    }
+                });
+            }
+
+            function deleteItem(event) {
+                isAdded = false;
+                const itemId = event.target.parentElement.id; // getting the id of the parent element to search in the array
+                todoList.some(function (todoItem, i) {
+                    if (todoItem.id === itemId) {
+                        todoList.splice(i, 1);
+                        renderList();
+                        return true;
+                    }
+                });
+            }
+
+            function saveTodo(e) {
+                localStorage.setItem("todoList", JSON.stringify(todoList));
+            }
+            function deleteTodo(e) {
+                localStorage.clear();
```

Comment:
@sauravkr818 Also, don't clear the entire localStorage, other sites may have their session data in there too, so use localStorage.removeItem()